marimo/_server/api/endpoints/ws.py

```diff
@@ -323,64 +366,50 @@ async def listen_for_messages() -> None:
             while True:
                 (op, data) = await self.message_queue.get()
                 try:
-                    await self.websocket.send_text(
-                        json.dumps(
-                            {
-                                "op": op,
-                                "data": data,
-                            },
-                            cls=WebComponentEncoder,
-                        )
+                    text = json.dumps(
+                        {
+                            "op": op,
+                            "data": data,
+                        },
+                        cls=WebComponentEncoder,
                     )
                 except TypeError as e:
                     # This is a deserialization error
                     LOGGER.error(
                         "Failed to send message to frontend: %s", str(e)
                     )
                     LOGGER.error("Message: %s", data)
+                    continue
+
+                try:
+                    await self.websocket.send_text(text)
+                except WebSocketDisconnect as e:
+                    self._on_disconnect(
+                        e,
+                        cleanup_fn=lambda: listen_for_disconnect_task.cancel(),
+                    )
+                except RuntimeError as e:
+                    # Starlette can raise a runtime error if a message is sent
+                    # when the socket is closed. In case the disconnection
+                    # error hasn't made its way to listen_for_disconnect, do
```

Comment:
I am not really sure if this case is needed. I occasionally saw `RuntimeError`s raised on `send_text` saying the socket was already closed